grails-app/services/org/broadinstitute/orsp/StorageProviderService.groovy

```diff
@@ -163,21 +163,22 @@ class StorageProviderService implements Status {
      * @param files The multipart files
      * @return
      */
-    StorageDocument saveMultipartFile(String displayName, String userName, String issueKey, String type, MultipartFile file) {
+    StorageDocument saveMultipartFile(String displayName, String userName, String issueKey, String type, MultipartFile file, Long consentCollectionLinkId) {
         StorageDocument document = new StorageDocument(
-                projectKey: issueKey,
+                projectKey: consentCollectionLinkId == null ? issueKey : null,
```

Comment:
This doesn't look right. This adds a layer of business logic to the method that doesn't belong here. If this method is handed both a `projectKey` and a `consentCollectionLinkId`, then this method should honor what was asked of it and insert both values as they were provided. It might not make logical sense from a larger application perspective, but that business logic should be elsewhere, not here where we're storing the document.